### PR TITLE
Migrate to esConsumes AlignmentProducerAsAnalyzer and MuonMillepedeTrackRefitter modules

### PR DESCRIPTION
**Pull request by hyunyong**
#### PR description:

Part of #31061.

Migrate to esConsumes ```AlignmentProducerAsAnalyzer``` and ```MuonMillepedeTrackRefitter``` modules

#### PR validation:

#33583 has validtated.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

No backport is needed.

### CODE REVIEW

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
Hi @hyunyong, can you please deal with the static analyzer error, too? You seem to be using a legacy module:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-abe914/16965/llvm-analysis/legacy-mod-sa.txt

**Comment by hyunyong** [comment]
> Hi @hyunyong, can you please deal with the static analyzer error, too? You seem to be using a legacy module:
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-abe914/16965/llvm-analysis/legacy-mod-sa.txt

Yes, I've changed that module.

**Comment by tvami** [comment]
> > Hi @hyunyong, can you please deal with the static analyzer error, too? You seem to be using a legacy module:
> > https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-abe914/16965/llvm-analysis/legacy-mod-sa.txt
> 
> Yes, I've changed that module.

Thanks, I'll trigger tests soon, but right now the IB seems to have problems.

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
Hi @cvuosalo , this failure seems to be a geometry issue, do you have any insights on this?

File: Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py [hunk 1]
```diff
@@ -76,28 +76,16 @@
...
-)
-GEMGeometryAlignmentProducerAsAnalyzer = cms.ESProducer("GEMGeometryESModule",
-    appendToDataLabel = cms.string('idealForAlignmentProducerBase'),
-    applyAlignment = cms.bool(False),
-    alignmentsLabel = cms.string(''),
-    fromDDD = cms.bool(True),
-    fromDD4Hep = cms.bool(False)
-)
+
+from Geometry.CMSCommonData.cmsIdealGeometryXML_cfi import *
```
Comment (cvuosalo) [inline Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py:80]: Why was this line added?

File: Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py [hunk 2]
```diff
@@ -76,28 +76,16 @@
...
+import Geometry.DTGeometryBuilder.dtGeometry_cfi
+DTGeometryAlignmentProducerAsAnalyzer = Geometry.DTGeometryBuilder.dtGeometry_cfi.DTGeometryESModule.clone()
+DTGeometryAlignmentProducerAsAnalyzer.appendToDataLabel = 'idealForAlignmentProducerBase'
+import Geometry.CSCGeometryBuilder.cscGeometry_cfi
+CSCGeometryAlignmentProducerAsAnalyzer = Geometry.CSCGeometryBuilder.cscGeometry_cfi.CSCGeometryESModule.clone()
+CSCGeometryAlignmentProducerAsAnalyzer.appendToDataLabel = 'idealForAlignmentProducerBase'
+import   Geometry.GEMGeometryBuilder.gemGeometry_cfi 
+GEMGeometryAlignmentProducerAsAnalyzer = Geometry.GEMGeometryBuilder.gemGeometry_cfi.gemGeometry.clone()
```
Comment (cvuosalo) [inline Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py:88]:
```suggestion
GEMGeometryAlignmentProducerAsAnalyzer = Geometry.GEMGeometryBuilder.gemGeometry_cff.gemGeometry.clone()
```

File: Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py [hunk 3]
```diff
@@ -76,28 +76,16 @@
...
+GEMGeometryAlignmentProducerAsAnalyzer.appendToDataLabel = 'idealForAlignmentProducerBase'
+GEMGeometryAlignmentProducerAsAnalyzer.fromDDD = False
+GEMGeometryAlignmentProducerAsAnalyzer.fromDD4Hep = False
```
Comment (cvuosalo) [inline Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py:91]: Lines 90 and 91 should be deleted. `gemGeometry.cff` takes care of these parameters.

**Comment by cvuosalo** [comment]
For workflow 11634.912, `step5_ALCA.py` needs to be checked with `edmConfigDump`. It should contain `DDCompactViewESProducer`. If it doesn't, that would explain the error.

**Comment by tvami** [comment]
> For workflow 11634.912, `step5_ALCA.py` needs to be checked with `edmConfigDump`. It should contain `DDCompactViewESProducer`. If it doesn't, that would explain the error.

I did this, indeed, `step5_ALCA.py`  does not contain `DDCompactViewESProducer`

**Comment by cvuosalo** [comment]
The Muon Geometry Constants require the `DDCompactViewESProducer`. However, prior to this PR, step5 did not seem to use the The Muon Geometry Constants. Why does this PR introduce dependence on the the Muon Geometry Constants?

File: Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py [hunk 4]
```diff
@@ -76,28 +76,16 @@
...
-    appendToDataLabel = cms.string('idealForAlignmentProducerBase'),
-    applyAlignment = cms.bool(False),
-    alignmentsLabel = cms.string(''),
-    fromDDD = cms.bool(True),
-    fromDD4Hep = cms.bool(False)
-)
+
+from Geometry.CMSCommonData.cmsIdealGeometryXML_cfi import *
+import Geometry.DTGeometryBuilder.dtGeometry_cfi
+DTGeometryAlignmentProducerAsAnalyzer = Geometry.DTGeometryBuilder.dtGeometry_cfi.DTGeometryESModule.clone()
```
Comment (cvuosalo) [inline Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py:82]:
```suggestion
DTGeometryAlignmentProducerAsAnalyzer = Geometry.DTGeometryBuilder.dtGeometryDB_cfi.DTGeometryESModule.clone()
```

File: Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py [hunk 5]
```diff
@@ -76,28 +76,16 @@
...
+DTGeometryAlignmentProducerAsAnalyzer.appendToDataLabel = 'idealForAlignmentProducerBase'
+import Geometry.CSCGeometryBuilder.cscGeometry_cfi
+CSCGeometryAlignmentProducerAsAnalyzer = Geometry.CSCGeometryBuilder.cscGeometry_cfi.CSCGeometryESModule.clone()
```
Comment (cvuosalo) [inline Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py:85]:
```suggestion
CSCGeometryAlignmentProducerAsAnalyzer = Geometry.CSCGeometryBuilder.cscGeometryDB_cfi.CSCGeometryESModule.clone()
```

File: Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py (hunk 2 above)
Comment (cvuosalo) [inline Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py:88]:
```suggestion
GEMGeometryAlignmentProducerAsAnalyzer = Geometry.GEMGeometryBuilder.gemGeometryDB_cfi.gemGeometry.clone()
```

**Comment by cvuosalo** [comment]
I am hoping with the changes I suggested that the Muon Geometry Constants will not be used and thus the `DDCompactView` will not be needed.

File: Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py (hunk 3 above)
Comment (hyunyong) [reply Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py:91]: ```gemGeometryDB_cff``` doesn't have the ```appendToDataLabel``` parameter.
So, keep this way or need to add ```appendToDataLabel```  at  ```gemGeometryDB_cff```.
What do you think?

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by cvuosalo** [comment]
+1

**Comment by tvami** [comment]
+alca

File: Alignment/MuonAlignmentAlgorithms/plugins/MuonMillepedeTrackRefitter.h [hunk 6]
```diff
@@ -13,9 +13,15 @@
 #define Alignment_MuonMillepedeTrackRefitter_MuonMillepedeTrackRefitter_H
 
 // Base Class Headers
-#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
```
Comment (srimanob) [inline Alignment/MuonAlignmentAlgorithms/plugins/MuonMillepedeTrackRefitter.h:16]: Could you please confirm if the private test is done with multicore, and you still get the memory in control, i.e. this module can run on multiple streams.

Comment (hyunyong) [reply Alignment/MuonAlignmentAlgorithms/plugins/MuonMillepedeTrackRefitter.h:16]: I've tried to validate and run on multiple streams but we never use this module recently and the cosmic dataset is not working with this module. 
I think we can clean up the millipede method in the muon alignment modules.
However, is it fine?

Comment (tvami) [reply Alignment/MuonAlignmentAlgorithms/plugins/MuonMillepedeTrackRefitter.h:16]: @hyunyong I'm not sure I understand "we can clean up the millipede method in the muon alignment modules.", can you please clarify? 
Also "the cosmic dataset is not working with this module", couldn't you just use some older data for the tests?

Comment (hyunyong) [reply Alignment/MuonAlignmentAlgorithms/plugins/MuonMillepedeTrackRefitter.h:16]: @tvami We don't have a plan to use this millipede method anymore. So, I think we can remove the module but I don't know the CMSSW policy.

Comment (tvami) [reply Alignment/MuonAlignmentAlgorithms/plugins/MuonMillepedeTrackRefitter.h:16]: @srimanob , @mmusich what do you think?

Comment (mmusich) [reply Alignment/MuonAlignmentAlgorithms/plugins/MuonMillepedeTrackRefitter.h:16]: If the class is not used it can just be deleted

Comment (tvami) [reply Alignment/MuonAlignmentAlgorithms/plugins/MuonMillepedeTrackRefitter.h:16]: Great! Then @hyunyong please go ahead with the removal

Comment (hyunyong) [reply Alignment/MuonAlignmentAlgorithms/plugins/MuonMillepedeTrackRefitter.h]: @tvami Yes, I've removed it.

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+alca

**Comment by tvami** [comment]
We don't expect any other changes with this PR, right?

**Comment by srimanob** [comment]
+Upgrade

**Comment by cvuosalo** [comment]
+1

**Comment by tvami** [comment]
@cms-sw/core-l2 I think there is a single file that made the PR require a signature from you, can you please review and sign? thanks!

**Comment by makortel** [comment]
+core

> I think there is a single file that made the PR require a signature from you, can you please review and sign? thanks!

Missed that, thanks for pinging.

**Comment by perrotta** [comment]
+1

**Comment by makortel** [comment]
(sorry for the noise if these are already being worked on)

Just to note that there are further calls to helper functions that call the deprecated `EventSetupRecord::get()` from `AlignmentProducerAsAnalyzer` as shown [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_12_0_X_2021-07-26-2300/slc7_amd64_gcc900/reports/eventsetuprecord-get.yaml). Summary of the call stacks
```
AlignmentProducerAsAnalyzer::beginRun(); AlignmentProducerBase::beginRunImpl(); IntegratedCalibrationBase::beginRun() virtual; SiPixelLorentzAngleCalibration::beginRun();

AlignmentProducerAsAnalyzer::accumulate(); AlignmentProducerBase::processEvent(); AlignmentAlgorithmBase::run() virtual; MillePedeAlignmentAlgorithm::run(); TrajectoryFactoryBase::trajectories() const virtual; BzeroReferenceTrajectoryFactory::trajectories() const;
AlignmentProducerAsAnalyzer::accumulate(); AlignmentProducerBase::processEvent(); AlignmentAlgorithmBase::run() virtual; MillePedeAlignmentAlgorithm::run(); TrajectoryFactoryBase::trajectories() const virtual; DualBzeroTrajectoryFactory::trajectories() const;
AlignmentProducerAsAnalyzer::accumulate(); AlignmentProducerBase::processEvent(); AlignmentAlgorithmBase::run() virtual; MillePedeAlignmentAlgorithm::run(); TrajectoryFactoryBase::trajectories() const virtual; DualTrajectoryFactory::trajectories() const;
AlignmentProducerAsAnalyzer::accumulate(); AlignmentProducerBase::processEvent(); AlignmentAlgorithmBase::run() virtual; MillePedeAlignmentAlgorithm::run(); TrajectoryFactoryBase::trajectories() const virtual; ReferenceTrajectoryFactory::trajectories() const;
AlignmentProducerAsAnalyzer::accumulate(); AlignmentProducerBase::processEvent(); AlignmentAlgorithmBase::run() virtual; MillePedeAlignmentAlgorithm::run(); TrajectoryFactoryBase::trajectories() const virtual; TwoBodyDecayTrajectoryFactory::trajectories() const;

AlignmentProducerAsAnalyzer::accumulate(); AlignmentProducerBase::processEvent(); AlignmentAlgorithmBase::run() virtual; MuonAlignmentFromReference::run(); MuonResidualsFromTrack::MuonResidualsFromTrack();
```

The report from CMSSW_12_0_X_2021-07-26-2300 shows also
```
AlignmentProducerAsAnalyzer::accumulate(); AlignmentProducerBase::processEvent(); AlignmentAlgorithmBase::run() virtual; MillePedeAlignmentAlgorithm::run(); TrajectoryFactoryBase::trajectories() const virtual; TwoBodyDecayTrajectoryFactory::trajectories() const; reco::TransientTrack::setES();
```
that was addressed in https://github.com/cms-sw/cmssw/pull/34584, but we don't have static analyzer reports for the CMSSW_12_0_X_2021-07-27-2300 IB yet.
